c/mptcphello.c

```diff
@@ -5,34 +5,52 @@
 #include <sys/socket.h>
 #include <stdbool.h>
 #include <netinet/in.h>
+#include <errno.h>
 
+// IPPROTO_MPTCP is defined in <netinet/in.h> on recent kernels only
 
 #ifndef IPPROTO_MPTCP
 #define IPPROTO_MPTCP 262
 #endif
 
 
 
+/*
+ * Create a Multipath TCP socket for the specified domain 
+ * If the creation fails because Multipath TCP is not supported
+ * on this system, falls back to regular TCP
+ */
+int socket_create(int domain) {
 
-bool use_mptcp = true;
+  // if true, try always enable Multipath TCP for TCP sockets
+  static bool use_mptcp = true;
+  int s;
+  if(use_mptcp) {
+    s = socket(domain, SOCK_STREAM, IPPROTO_MPTCP);
+    if(s==-1 && ( errno==EPROTONOSUPPORT || errno==ENOPROTOOPT) ) {
+      // Multipath TCP is not supported on this system
+      use_mptcp = false;
```

Comment:
maybe good to re-add a clear error message?

    fprintf(stderr, "Unable to create an MPTCP socket: "
            "your kernel probably don't support MPTCP yet, "
            "falling back to TCP for this socket and all the next ones.\n");